### DISCUSSION TITLE
[Question] Today's lecture (slide 7, page 20)

### DISCUSSION THREAD

**Discussion by quanxi1** [prosyslab-classroom/cs424-program-reasoning#276] [Q&A]
Name: Taehoon Lee 

Hello, 
I have a question on today's lecture content. Actually I already asked this in the last part of the lecture, but I couldn't totally get it. 
I'm totally new to all these concepts, so I think any advice by students would help me a lot. 

The slide denotes that: 
> Meanig of Hoare triple {P} S {Q} : 
> - If S is executed in a state satisfying P and if the execution of S terminates 
> - Then, the program state after S terminates satisfies Q 

For the 4th example on slide 20,
> {false} x := y + 3 {y=0 && x=12}

I remember that professor explained this is valid because there's no assumtion under which the assignment derives the postcondition, 
in other words(as far as I understood),
> {y=0 && x=12  [(y+3) / x]} x := y + 3 {y=0 && x=12}

 is always valid by definition
and the P term here is indeed false because it is unsatisfiable after substitution. 

I understood this, but I still wonder why this explanation is not acceptable : 
The triple holds iff (if P is true and S terminates, then Q). In the given triple, P is false; which means the triple holds trivially.<br>

I mean, 
> {false} x := 1 {x = 1}

is this triple valid or not?

**Comment by KihongHeo** [comment]
Hi. Thanks for the good question.
Sorry, my mistake. You are right. If P is false, the triple is valid.

> {false} x := 1 {x = 1}

This triple can be proven by using the assignment rule and precondition strengthening (i.e., false => true).
